src/common/rules.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*jshint forin:true, noarg:false, noempty:true, eqeqeq:true, bitwise:true,
+  strict:true, undef:true, curly:false, browser:true,
+  unused:true, indent:2, maxerr:50, devel:true, node:true, boss:true, white:true,
+  globalstrict:true, nomen:false, newcap:true, esnext: true, moz: true,
+  asi: true  */
+
+/*global module */
+
+function isRe (thing) {
+  return Object.prototype.toString.call( thing ) == "[object RegExp]"
+}
+
+function matchRulePart(subObj, expected) {
```

Comment:
only regex and 'everything else'.  No functions.

NOT SUPPORTED:
- thing.level1.level2